apps/verticalhome/js/pin_apps.js

```diff
@@ -36,25 +36,50 @@
       this.render();
     },
 
+
     render: function() {
-      this.element.removeEventListener("click", this);
+
+      var self = this;
+      function setName() {
```

Comment:
why this function called 'setName'? it just "calculates" the appropriate name value and not sets it to anywhere.